### PR TITLE
methods-modular_programs-medium-q4

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

File: src/methods/modular_programs/medium/q4/CharaComb.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package methods.modular_programs.medium.q4;
+
```
Comment (KaiKhademi) [inline src/methods/modular_programs/medium/q4/CharaComb.java:2]: Do not merge

Comment (Hedgemon4) [reply src/methods/modular_programs/medium/q4/CharaComb.java:2]: Why is this do not merge?

File: src/methods/modular_programs/medium/q4/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        return Stream.of(99.99999, 0.0, 0.1, 99.0, 34.12, 55.2, 3.1);
+    }
+    static Stream<Double> inputProvider() {
+        return Stream.of(100.0, 100.1, 200.0, 1231.123, 5555.11, 99.99999, 0.0, 0.1, 99.0, 34.12, 55.2, 3.1);
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputAbove100Provider")
+    public void correctsOutputAbove100(double temp) throws InvalidClauseException {
+        TestOption.incorrectStructureErrorMessage = "Your program does not print the correct message when temperature is above 100 oF";
```
Comment (Hedgemon4) [inline src/methods/modular_programs/medium/q4/MainTest.java:48]:
```suggestion
        TestOption.incorrectStructureErrorMessage = "Your program does not print the correct message when the temperature is above 100 degrees.";
```

File: src/methods/modular_programs/medium/q4/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        runWithInput(temp + "", new Clause[]{
+                new StringLiteral("You have a fever.")
+        });
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputBelow100Provider")
+    public void correctsOutputBelow100(double temp) throws InvalidClauseException {
+        TestOption.incorrectStructureErrorMessage = "Your program does not print the correct message when temperature is below 100 oF";
```
Comment (Hedgemon4) [inline src/methods/modular_programs/medium/q4/MainTest.java:57]:
```suggestion
        TestOption.incorrectStructureErrorMessage = "Your program does not print the correct message when the temperature is below 100 degrees.";
```

**Review by Hedgemon4 — APPROVED** [review]

**Pull request merged**